### PR TITLE
Removed PodStatus.Host

### LINKED ISSUE #6165: Eliminate PodStatus.Host in v1beta3
PodSpec.Host sets the node that the pod should be running on. PodStatus.Host should be eliminated in v1beta3; it should be populated from PodSpec.Host in v1beta1/2.

/cc @bgrant0607

### PR DESCRIPTION
**Pull request by piosz**
Fixes #6165

### CODE REVIEW

**Comment by piosz** [comment]
cc @yujuhong

File: pkg/api/validation/validation.go [hunk 1]
```diff
@@ -769,8 +769,8 @@ func ValidatePodStatusUpdate(newPod, oldPod *api.Pod) errs.ValidationErrorList {
 	allErrs = append(allErrs, ValidateObjectMetaUpdate(&oldPod.ObjectMeta, &newPod.ObjectMeta).Prefix("metadata")...)
 
 	// TODO: allow change when bindings are properly decoupled from pods
-	if newPod.Status.Host != oldPod.Status.Host {
-		allErrs = append(allErrs, errs.NewFieldInvalid("status.host", newPod.Status.Host, "pod host cannot be changed directly"))
+	if newPod.Spec.Host != oldPod.Spec.Host {
+		allErrs = append(allErrs, errs.NewFieldInvalid("status.hostIP", newPod.Status.HostIP, "pod host cannot be changed directly"))
```
Comment (wojtek-t) [inline pkg/api/validation/validation.go]: This should be spec.Host instead of status.hostIP

Comment (piosz) [inline pkg/api/validation/validation.go]: Done

**Comment by brendandburns** [comment]
LGTM.

**Comment by bgrant0607** [comment]
LGTM. Did you run e2e?

**Comment by bgrant0607** [comment]
We'll regenerate the swagger later.

**Comment by piosz** [comment]
Yes, but I need to rerun them due to an issue not related to change. I'll let you know once they passes.

**Comment by piosz** [comment]
Tests that are not passing:
* test/e2e/networking.go
* test/e2e/pd.go

Looking like flakiness, but please merge carefully. Alternatively I can rerun tests tomorrow.

**Comment by bgrant0607** [comment]
Merging.

**Comment by piosz** [comment]
Thanks for review!

**Pull request merged**